gtsfm/evaluation/visualize_benchmark_comparison.py

```diff
@@ -0,0 +1,254 @@
+
+
+def colorscale_from_list(alist, name):
+    # Defines a colormap, and the corresponding Plotly colorscale from the list alist
+    # alist=the list of basic colors
+    # name is the name of the corresponding matplotlib colormap
+
+    cmap = LinearSegmentedColormap.from_list(name, alist)
+    colorscale = colormap_to_colorscale(cmap)
+    return cmap, colorscale
+
+
+def plot_colored_table(X: List[str], Y: List[str], Z: np.ndarray, table_name: str) -> None:
+    """Create an annotated heatmap.
+
+    Args:
+        X: labels for each column (column names).
+        Y: labels for each row (row names).
+        Z: 2d matrix, representing percentage changes from value for a metric on the master branch.
+    """
+
+    # clip Z to -100% and +100%, and then scale result to a linear scale [0,100]
+    Z_clipped = np.clip(Z, a_min=MIN_RENDERABLE_PERCENT_CHANGE, a_max=MAX_RENDERABLE_PERCENT_CHANGE)
+
+    redgreen = [RED_HEX, PALE_YELLOW_HEX, GREEN_HEX]
+    fin_cmap, fin_cs = colorscale_from_list(redgreen, "fin_cmap")
+    trace = go.Heatmap(
+        z=Z_clipped,
+        x=X,
+        y=Y,
+        colorscale=fin_cs,  # fin_asymm_cs,
+        zmin=-MIN_RENDERABLE_PERCENT_CHANGE,
+        zmax=MAX_RENDERABLE_PERCENT_CHANGE
+        # colorbar=dict(thickness=20, tickvals=[-20+k*5 for k in range(5)], ticklen=4)
+    )
+
+    layout = go.Layout(
+        title="Percentage Change",
+        font=Font(family="Balto, sans-serif", size=12, color="rgb(68,68,68)"),
+        showlegend=False,
+        xaxis=XAxis(title="", showgrid=True, side="top", tickangle=-45),
+        yaxis=YAxis(
+            title="",
+            autorange="reversed",
+            showgrid=True,
+            # autotick=False,
+            # dtick=1
+        ),
+        autosize=False,
+        height=HEATMAP_HEIGHT,
+        width=HEATMAP_WIDTH,
+        margin=Margin(l=135, r=40, b=85, t=170),
+    )
+
+    fig = go.Figure(data=go.Data([trace]), layout=layout)
+
+    annotations = go.Annotations()
+    alist = list(Z)
+
+    num_rows, num_cols = Z.shape
+    for i in range(num_rows):
+        for j in range(num_cols):
+            annotations.append(
+                go.Annotation(
+                    text=str(np.round(Z[i, j], 1)) + "%",
+                    x=X[j],
+                    y=Y[i],
+                    xref="x1",
+                    yref="y1",
+                    font=dict(color="rgb(25,25,25)"),
+                    showarrow=False,
+                )
+            )
+    fig["layout"].update(annotations=annotations)
+    return fig.to_html(full_html=False, include_plotlyjs="cdn")
+
+
+def generate_dashboard(curr_master_dirpath: str, new_branch_dirpath: str) -> None:
+    """Generate a dashboard showing a visual representation of the diff against master on all benchmarks.
+
+    Args:
+        curr_master_dirpath
+        new_branch_dirpath
+    """
+
+    html_path = "visual_comparison_dashboard.html"
```

Comment:
should this be made a constant or an argument?

Reply:
I made it a global constant to make it clearer where the file would be saved